server/data/user.go

```diff
@@ -0,0 +1,126 @@
+package data
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/jackc/pgx/v5"
+	"golang.org/x/crypto/bcrypt"
+)
+
+type User struct {
+	Username    string `db:"username"`
```

Comment:
There's no reason to include the `Username` field here. Maybe we could rename this struct `UserProfile` and remove the username field so that it matches the project documentation?

Reply:
I do agree it is an additional transaction we could save up. We are already querying by the username so you would already have it 😅

If we maybe need it in the future for searching users we would need to fetch an array of UserProfiles in which we need a "unique identifier" for entering into their profiles or something